File: 06_이진탐색/06_부품찾기.py
def binarySearch(nArr, start, end, target) :
    if start > end :
        return False
    mid = (start + end) // 2
    if nArr[mid] == target :
        return True
    elif nArr[mid] < target :
        return binarySearch(nArr, mid + 1, end, target)
    elif nArr[mid] > target :
        return binarySearch(nArr, start, mid - 1, target)


n = int(input())
nArr = list(map(int, input().split()))
nArr.sort()

m = int(input())
mArr = list(map(int, input().split()))
for i in range(m) :
    if binarySearch(nArr, 0, n-1, mArr[i]) == True:
        print("yes", end=' ')
    else :
        print("no", end=' ')

'''

5
8 3 7 9 2
3
5 7 9

'''

File: 06_이진탐색/06_부품찾기.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(void){

    int n, m;
    vector<int> nArr;

    cin >> n;
    for (int i = 0; i < n; i++)
    {
        int temp;
        cin >> temp;
        nArr.push_back(temp);
    }
    // 이진 탐색을 위해 정렬
    sort(nArr.begin(), nArr.end());

    cin >> m;
    for (int i = 0; i < m; i++)
    {
        int temp;
        cin >> temp;
        // 있으면?
        if(binary_search(nArr.begin(), nArr.end(), temp)) {
            cout << "yes ";
        }  
        // 없으면?
        else {
            cout << "no ";
        }
    }
    


    return 0;
}

/*

5
8 3 7 9 2
3
5 7 9

*/